wp/plugin/lib/analysis.ml

```diff
@@ -138,25 +138,48 @@ let sp (arch : Arch.t) : (Comp.comparator * Comp.comparator) option =
            memory for the x86_64 architecture.\n%!";
     None
 
+(* Create list of variables from string names. *)
+let create_vars (l : string list) : Bap.Std.Var.Set.t =
+  List.map l ~f:(fun var -> Var.create var reg64_t) |> Bap.Std.Var.Set.of_list
```

Comment:
Hardcoding the variables as `reg64_t` kinda worries me, especially in the case where our input registers happen to be 32-bit. Is there a way to determine the width of the register? Perhaps the Arch or Target modules have something? I'm not really sure.

Reply:
There's nothing as far as I can see in `Arch` or `Target` that fetches the register size. The two solutions I see are:
- Use the width of the sp (which I can snag off of `Target`) as the width
- Use the word size (which I can grab from `Arch`) as the width.

I'm not sure which would be preferable here.

Reply:
Following @fortunac 's [observation](http://binaryanalysisplatform.github.io/bap/api/master/bap-x86-cpu/X86_cpu/AMD64/index.html) of `Target.CPU.gpr`, I've updated `create_vars` to either return a general purpose register if its name matches the input, or fail if the name is not in the set of general purpose registers for the program's architecture.